sql/stmtdiagnostics: TestDiagnosticsRequest failed

sql/stmtdiagnostics.TestDiagnosticsRequest [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20540889?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20540889?buildTab=artifacts#/) on master @ [9f71acc90da1e535be852b798899685d4fd668da](https://github.com/cockroachdb/cockroach/commits/9f71acc90da1e535be852b798899685d4fd668da):


```
=== RUN   TestDiagnosticsRequest
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/85a915e05ca8994a6ad8cbe3d0421959/logTestDiagnosticsRequest2932946513
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:154: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    statement_diagnostics_test.go:544: -- test log scope end --
--- FAIL: TestDiagnosticsRequest (6.19s)
=== RUN   TestDiagnosticsRequest/timeout
    statement_diagnostics_test.go:177: error executing query="RESET statement_timeout;" args=[]: pq: query execution canceled due to statement timeout
--- FAIL: TestDiagnosticsRequest/timeout (0.33s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #152944 sql/stmtdiagnostics: TestDiagnosticsRequest failed [C-test-failure O-robot T-sql-queries branch-release-25.3]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDiagnosticsRequest.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-54960

--- a/pkg/sql/stmtdiagnostics/statement_diagnostics_test.go
+++ b/pkg/sql/stmtdiagnostics/statement_diagnostics_test.go
@@ -173,8 +173,12 @@ func TestDiagnosticsRequest(t *testing.T) {
 		require.True(t, strings.Contains(err.Error(), sqlerrors.QueryTimeoutError.Error()))
 
 		// Reset the stmt timeout so that it doesn't affect the query in
-		// checkCompleted.
-		runner.Exec(t, "RESET statement_timeout;")
+		// checkCompleted. Wrap it in a SucceedsSoon in case RESET query itself
+		// times out.
+		testutils.SucceedsSoon(t, func() error {
+			_, err = db.Exec("RESET statement_timeout;")
+			return err
+		})
 		checkCompleted(reqID)
 	})
 

